synapse/lib/scrape.py

```diff
@@ -463,3 +547,87 @@ def scrape(text, ptype=None, refang=True, first=False):
 
     for info in contextScrape(text, form=ptype, refang=refang, first=first):
         yield info.get('form'), info.get('valu')
+
+def _spawn_scrape(sock, text, form=None, refang=True, first=False): # pragma: no cover
+    '''
+    Multiprocessing target for scraping text.
+    '''
+    try:
+        for info in _contextScrape(text, form=form, refang=refang, first=first):
+            sock.sendall(s_msgpack.en((True, info)))
+
+        sock.sendall(s_msgpack.en((True, None)))
+
+    except Exception as e:
+        mesg = s_common.retnexc(e)
+        sock.sendall(s_msgpack.en(mesg))
+
+async def contextScrapeAsync(text, form=None, refang=True, first=False):
+    '''
+    Scrape types from a blob of text and yield info dictionaries, potentially in a spawned process.
+
+    Args:
+        text (str): Text to scrape.
+        form (str): Optional form to scrape. If present, only scrape items which match the provided form.
+        refang (bool): Whether to remove de-fanging schemes from text before scraping.
+        first (bool): If true, only yield the first item scraped.
+
+    Notes:
+        The dictionaries yielded by this function contains the following keys:
+
+            match
+                The raw matching text found in the input text.
+
+            offset
+                The offset into the text where the match was found.
+
+            valu
+                The resulting value.
+
+            form
+                The corresponding form for the valu.
+
+    Returns:
+        (dict): Yield info dicts of results.
+    '''
+    if len(text) < SCRAPE_SPAWN_LENGTH:
+        for info in _contextScrape(text, form=form, refang=refang, first=first):
+            yield info
+        return
+
+    link00, sock00 = await s_link.linksock()
+
+    try:
+        async with await s_base.Base.anit() as scope:
+
+            todo = s_common.todo(_spawn_scrape, sock00, text, form=form, refang=refang, first=first)
+            scope.schedCoro(s_coro.spawn(todo, log_conf=s_common._getLogConfFromEnv()))
+
+            while (mesg := await link00.rx()) is not None:
+
+                info = s_common.result(mesg)
+                if info is None:
+                    return
+
+                yield info
+
+    finally:
+        sock00.close()
+        await link00.fini()
```

Comment:
if we use the link as a context manager we don't have to call fini on him

Reply:
This link does not need to be fini'd here

Reply:
forgot to delete that one, fixed